src/org/zaproxy/zap/view/widgets/WritableFileChooser.java

```diff
@@ -24,6 +24,11 @@ public WritableFileChooser(File currentDirectory) {
 	@Override
 	public void approveSelection() {
 		File selectedFile = getSelectedFile();
+		
+		if (selectedFile.length() > selectedFile.getUsableSpace()) {
```

Comment:
I'm not sure this check would prevent the errors about disk space (when writing the report or other info).
The selected file might not exist, which would return 0L for both methods, even if it does exist it's not necessarily the size that the final file will have (in most cases the data was not yet generated so the final size is unknown, at least by this class).
I'm not sure the issue is fixable in this class.

This is what get from those calls for a file that exists and another that doesn't:
```
/exists           length: 511906 getUsableSpace: 16297455616
/does_not_exist   length: 0      getUsableSpace: 0
```